spring-cloud-starter-stream-source-twitterstream/src/main/java/org/springframework/cloud/stream/app/twitterstream/source/TwitterStreamMessageProducer.java

```diff
@@ -16,55 +16,68 @@
 
 package org.springframework.cloud.stream.app.twitterstream.source;
 
-import java.net.URI;
-
 import org.springframework.social.support.URIBuilder;
 import org.springframework.social.twitter.api.impl.TwitterTemplate;
 import org.springframework.util.StringUtils;
 
+import java.net.URI;
+
 /**
- *  {@link org.springframework.integration.core.MessageProducer} implementation to send Twitter stream messages.
+ * {@link org.springframework.integration.core.MessageProducer} implementation to send Twitter stream messages.
  *
  * @author Ilayaperumal Gopinathan
  * @author Gary Russell
  */
 class TwitterStreamMessageProducer extends AbstractTwitterInboundChannelAdapter {
 
-	private static final String API_URL_BASE = "https://stream.twitter.com/1.1/statuses/";
+    private static final String API_URL_BASE = "https://stream.twitter.com/1.1/statuses/";
+
+    private final TwitterStreamProperties twitterStreamProperties;
+
+    TwitterStreamMessageProducer(TwitterTemplate twitterTemplate, TwitterStreamProperties twitterStreamProperties) {
+        super(twitterTemplate);
+        this.twitterStreamProperties = twitterStreamProperties;
+    }
+
+    @Override
+    protected URI buildUri() {
+        TwitterStreamType streamType = this.twitterStreamProperties.getStreamType();
+        URIBuilder b = URIBuilder.fromUri(API_URL_BASE + streamType.getPath());
+        //TODO: Support all the available properties
+        if (streamType == TwitterStreamType.FILTER) {
+            if (StringUtils.hasText(this.twitterStreamProperties.getFollow())) {
+                b.queryParam("follow", this.twitterStreamProperties.getFollow());
+            }
 
-	private final TwitterStreamProperties twitterStreamProperties;
+            if (StringUtils.hasText(this.twitterStreamProperties.getTrack())) {
+                b.queryParam("track", this.twitterStreamProperties.getTrack());
+            }
 
-	TwitterStreamMessageProducer(TwitterTemplate twitterTemplate, TwitterStreamProperties twitterStreamProperties) {
-		super(twitterTemplate);
-		this.twitterStreamProperties = twitterStreamProperties;
-	}
+            if (StringUtils.hasText(this.twitterStreamProperties.getLocations())) {
+                b.queryParam("locations", this.twitterStreamProperties.getLocations());
+            }
+        } else {
```

Comment:
An `else` on a new line.
Doesn't `language` used in the `filter` stream as well?

Reply:
I have change the else.
According to the documentation at https://developer.twitter.com/en/docs/tweets/filter-realtime/api-reference/post-statuses-filter.html `language` is not an official option.